Program.cs

```diff
@@ -29,13 +29,17 @@
 	{
 		var oldProcessId = process.Id;
 		var retryCount = 0;
-		while (retryCount < 2)
+		while (retryCount < 3)
 		{
 			var matchingProcesses = Process.GetProcessesByName("BF1942");
 			if (matchingProcesses.Length == 1)
 			{
 				process = matchingProcesses[0];
-				Console.WriteLine($"Process has changed ({oldProcessId} -> {process.Id})");
+				Console.WriteLine($"[Borderless1942]: [{DateTime.Now:yyyy-MM-dd - hh:mm:ss tt}] [BF1942 Process Has Changed] [{oldProcessId} -> {process.Id}]");
+				if (process.HasExited && process.MainWindowHandle == IntPtr.Zero)
+				{
+					break;
+				}
```

Comment:
OK, after debugging `Process.GetProcessesByName()`, I've been able to separately confirm that it can return closed processes. Here's what I think we should do:
- Drop the `process.MainWindowHandle` check
- Move the if-statement before the log output

Reply:
Hey @Turnerj,

Below I've outlined the latest changes I've made and the steps used in testing them.
Borderless1942 appears to be functioning correctly with the latest round of changes.

**Code changes:**
I made the requested changes to the body of "if (matchingProcesses.Length == 1)".


**Debugging output:**
Below is the debug console output from the latest round of code changes.

[Borderless1942]: [2022-08-27 - 08:39:34 AM] [BF1942 Process Has Started] [47108]
When Battle of Britain loaded, BF1942 did not create a new process for the first map in the playlist.

[Borderless1942]: [2022-08-27 - 08:40:57 AM] [BF1942 Process Has Changed] [47108 -> 46212]
When Battle of Midway loaded, BF1942 created the process above.

[Borderless1942]: [2022-08-27 - 08:42:01 AM] [BF1942 Process Has Changed] [46212 -> 36248]
When Battle of the Bulge loaded, BF1942 created the process above.

[Borderless1942]: [2022-08-27 - 08:42:31 AM] [BF1942 Process Has Exited]
When I hit ESC and closed the game from the main menu, Borderless1942 closed and output the console messages above.


**Testing Steps:**
Under Windows 11, I ran the following tests in a local COOP game using the following steps.

1. I opened a Terminal window in the root of my BF1942 install and ran ".\Borderless1942.exe" so that I would be able to see all of the console messages.

2. I created a local COOP mode playlist with the maps below.
Battle of Britain
Battle of Midway
Battle of the Bulge

3. To switch to the next map in the playlist, I used the in game map vote command by holding Insert and selecting the next map in the list.

4. As expected, BF1942 created a new process every time the map was changed except for when Battle of Britain loaded.

5. After Battle of the Bulge loaded, I hit ESC and then closed the game without navigating to any other parts of the menu.